# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mfbt/Atomics.h
# Commit: e87a5169f668
# Full Hash: e87a5169f66869fc200324bb4e0cd9a33583c5f9
# Author: stransky <stransky@redhat.com>
# Date: 2024-12-17 17:00:49
# Regressor Bug: 1934497
# File Overlap Count: 2
# Description:
#   Bug 1934497 [Wayland] Implement wl_surface abstraction as WaylandSurface r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230847
# ==============================================================================

diff -r 84186392210d -r e87a5169f668 mfbt/Atomics.h
--- a/mfbt/Atomics.h	Tue Dec 17 05:59:59 2024 +0000
+++ b/mfbt/Atomics.h	Tue Dec 17 05:59:59 2024 +0000
@@ -512,6 +512,27 @@
   Atomic(Atomic& aOther) = delete;
 };
 
+/**
+ * Atomic<T> implementation for double type.
+ */
+template <MemoryOrdering Order>
+class Atomic<double, Order> : protected detail::AtomicBase<double, Order> {
+  typedef typename detail::AtomicBase<double, Order> Base;
+
+ public:
+  constexpr Atomic() : Base() {}
+  explicit constexpr Atomic(double aInit) : Base(aInit) {}
+
+  operator double() const {
+    return double(Base::Intrinsics::load(Base::mValue));
+  }
+
+  double operator=(double aVal) { return Base::operator=(aVal); }
+
+ private:
+  Atomic(Atomic& aOther) = delete;
+};
+
 // Relax the CPU during a spinlock.  It's a good idea to place this in a
 // spinlock so that the CPU doesn't pipeline the loop otherwise flushing the
 // pipeline when the loop finally breaks can be expensive.